odoo_tools/cli/migrate_db.py

```diff
@@ -0,0 +1,597 @@
+        is_done = True
+        return is_done
+    container_dump_path = _get_db_prod_fixed_dump_path(in_container=True)
+    mount_opts = f"-v {ctx.obj['store_path']}:/{ctx.obj['container_store_path']}"
+    try:
+        _run_docker_compose_cmd(
+            f"run {mount_opts} --rm odoo pg_dump -b -Fc -d {db_name} "
+            f"-f {container_dump_path}"
+        )
+    except subprocess.CalledProcessError:
+        print("💥")
+        raise
+    print("✅", end="")
+    return True
+
+
+@click.pass_context
+def migrate_odoo(ctx):
+    upgraded_zip_path = ctx.obj["store_path"].joinpath("upgraded.zip")
+    if ctx.params["restart"] and upgraded_zip_path.exists():
+        upgraded_zip_path.unlink()
+    if upgraded_zip_path.exists():
+        print("ℹ️  (skipped: upgraded.zip file already exists)", end="")
+        return False
+    prod_dump_path = _get_db_prod_fixed_dump_path()
+    script_path = ctx.obj["store_path"].joinpath("odoo_upgrade.py")
+    urlretrieve(ODOO_UPGRADE_SCRIPT, script_path)
+    os.chdir(ctx.obj["store_path"])
+    cmd = (
+        f"TMPDIR=$(mktemp -d) /usr/bin/env python3 {script_path} production "
+        f"-c {ctx.obj['contract_number']} "
+        f"--dump {prod_dump_path} "
+        f"-t {ctx.obj['target_version']} "
+        f"--no-restore "
+    )
+    env_file_path = ctx.obj["odoo_upgrade_env_path"]
+    if env_file_path.exists():
+        cmd += f"--env-file {env_file_path}"
+    ret = subprocess.run(cmd, shell=True, capture_output=True)
+    if ret.returncode:
+        print("💥 FAILED 💥")
+        log_path = ctx.obj["store_path"].joinpath("upgrade.log")
+        raise SystemExit(f"=> Check logs located at {log_path}")
+    print("✅", end="")
+    return True
+
+
+@click.pass_context
+def restore_odoo_migrated(ctx):
+    # Check if Odoo S.A. database is already restored
+    db_name = ctx.obj["db_odoo_migrated"]
+    if _db_exists(db_name) and not ctx.params["restart"]:
+        print("ℹ️  (skipped: Odoo S.A. migrated database already restored)", end="")
+        return False
+    # Unzip upgraded.zip archive
+    upgraded_zip_path = ctx.obj["store_path"].joinpath("upgraded.zip")
+    upgraded_dir_path = ctx.obj["store_path"].joinpath("upgraded")
+    assert upgraded_zip_path.exists()
+    if not upgraded_dir_path.exists():
+        with zipfile.ZipFile(upgraded_zip_path) as upgraded:
+            upgraded.extractall(path=upgraded_dir_path)
+    # Restore Odoo S.A. migrated database
+    container_dump_sql_path = ctx.obj["container_store_path"].joinpath(
+        "upgraded", "dump.sql"
+    )
+    mount_opts = f"-v {ctx.obj['store_path']}:/{ctx.obj['container_store_path']}"
+    try:
+        _run_docker_compose_cmd(f"run --rm odoo dropdb --if-exists {db_name}")
+        _run_docker_compose_cmd(f"run --rm odoo createdb {db_name}")
+        _run_docker_compose_cmd(
+            f"run {mount_opts} -T --rm odoo psql -d {db_name} "
+            f"-f {container_dump_sql_path}"
+        )
```

Comment:
This one could return an error code while the dump is actually well restored (thanks @vvrossem for testing):
```
subprocess.CalledProcessError: Command 'docker compose --project-directory /home/vvanrossem/camptocamp/customer_odoo run -v 17.0.0.0.0://migrate-db-volume -T --rm odoo psql -d customer_621-20250619-122444_odoo_migrated -f /migrate-db-volume/upgraded/dump.sql' returned non-zero exit status 1.
```

Reply:
Should be fixed (no way to test currently, so :crossed_fingers: )